src/main/java/com/connexta/search/index/IndexManagerImpl.java

```diff
@@ -39,24 +41,25 @@ public void index(
     try {
       idAlreadyExists = crudRepository.existsById(productId);
     } catch (final Exception e) {
-      throw new IndexException("Unable to query index", e);
+      throw new IndexException(INTERNAL_SERVER_ERROR, "Unable to query index", e);
     }
     if (idAlreadyExists) {
-      throw new IndexException("Product already exists. Overriding is not supported");
+      throw new IndexException(
+          INTERNAL_SERVER_ERROR, "Product already exists. Overriding is not supported");
     }
 
     final Index index;
     try {
       index = new Index(productId, IOUtils.toString(inputStream, StandardCharsets.UTF_8));
     } catch (IOException e) {
-      throw new IndexException("Unable to convert InputStream to String", e);
+      throw new IndexException(INTERNAL_SERVER_ERROR, "Unable to convert InputStream to String", e);
     }
 
     log.info("Attempting to index product id {}", productId);
     try {
       crudRepository.save(index);
     } catch (final Exception e) {
-      throw new IndexException("Unable to save index", e);
+      throw new IndexException(INTERNAL_SERVER_ERROR, "Unable to save index", e);
```

Comment:
✏️ Create a new subclass of `IndexException`, maybe `DatabaseCommunicationErrorIndexException` for this line and the existsById check.

Reply:
I'm going to hold off on new exceptions for now. If IndexException proves too generic, we can add more exception classes.